### PR TITLE
Retry queries in events by tag stage

### PR DESCRIPTION
**Pull request by chbatey**
Refs #833

### CODE REVIEW

File: core/src/main/scala/akka/persistence/cassandra/Retries.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.concurrent.duration.FiniteDuration
+import scala.util.control.NonFatal
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[cassandra] object Retries {
```
Comment (chbatey) [inline core/src/main/scala/akka/persistence/cassandra/Retries.scala:19]: Simplified version of RetrySupport that also allows logging of message for each failure

File: core/src/main/resources/reference.conf [hunk 2]
```diff
@@ -341,6 +341,24 @@ akka.persistence.cassandra {
     # before the stream starts emitting events.
     offset-scanning-period = 200ms
 
+    # Settings for retrying database queries. It can still be wrapped in an RestartSource
+    # for full restarts on failure. Query retires are done within the stage because:
+    # * Lighter weight than a full restart. A restart requires scanning for the current tag pid sequence numbers
```
Comment (patriknw) [inline core/src/main/resources/reference.conf:346]: looks stange with the `# *`, `# -` would be better here

File: core/src/main/scala/akka/persistence/cassandra/EventsByTagSettings.scala [hunk 3]
```diff
@@ -27,6 +27,12 @@ import com.typesafe.config.Config
   case object Max extends Period
   case class Fixed(duration: FiniteDuration) extends Period
 
+  final case class RetrySettings(
+      retries: Int,
+      minDuration: FiniteDuration,
+      maxDuration: FiniteDuration,
+      randomFactor: Double)
```
Comment (patriknw) [inline core/src/main/scala/akka/persistence/cassandra/EventsByTagSettings.scala:34]: Latest Akka includes `akka.stream.RestartSettings`. Maybe we can use it here?

**Review by patriknw — APPROVED** [review]
LGTM

File: core/src/main/scala/akka/persistence/cassandra/EventsByTagSettings.scala (hunk 3 above)
Comment (chbatey) [reply core/src/main/scala/akka/persistence/cassandra/EventsByTagSettings.scala:34]: Okay. We don't have a reset as it is for a single query but we can still use it and ignore that

Comment (chbatey) [reply core/src/main/scala/akka/persistence/cassandra/EventsByTagSettings.scala:34]: Actually, there's no ctr that works for us

**Pull request merged**